lib/cocoapods/validator.rb

```diff
@@ -28,8 +28,12 @@ class Validator
     #         the Source URLs to use in creating a {Podfile}.
     #
     def initialize(spec_or_path, source_urls)
-      @source_urls = source_urls.map { |url| config.sources_manager.source_with_name_or_url(url) }.map(&:url)
       @linter = Specification::Linter.new(spec_or_path)
+      @source_urls = if !@linter.spec || @linter.spec.dependencies.empty?
```

Comment:
@benasher44 if I understood correctly, this is something already done with `has_dependencies?` and probably it's not enough, because map starts from `source_urls` (which defaults to master specs repo) and down the road will clone the master repo anyway, no matter what sources returns. Think about someone that wants to lint a Pod without dependencies and from a clean install (i.e. CI)